flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackendLoader.java

```diff
@@ -391,8 +393,8 @@ private static StateBackend loadChangelogStateBackend(
             Constructor<? extends DelegatingStateBackend> constructor =
                     Class.forName(CHANGELOG_STATE_BACKEND, false, classLoader)
                             .asSubclass(DelegatingStateBackend.class)
-                            .getConstructor(StateBackend.class);
-            return constructor.newInstance(backend);
+                            .getConstructor(StateBackend.class, StateChangelogWriterFactory.class);
+            return constructor.newInstance(backend, new InMemoryStateChangelogWriterFactory());
```

Comment:
Yes, it is hacky. We discussed it previously and [planned](https://github.com/apache/flink/pull/15200#discussion_r594669021) to implement separately in FLINK-21804 because of the time constraints.

However, I realized that another solution without modifying `StateBackendLoader` would better (passing the writer factory to backend.createKeyedBackend methods). So I'll remove this change alltogether.